src/Snowcap/Emarsys/Client.php

```diff
@@ -264,6 +264,24 @@ public function updateContact(array $data)
     }
 
     /**
+     * Updates one or more contacts/recipients, identified by an external ID. If the contact does not exist in the
+     * database, it is created.
+     *
+     * @param array $data
+     * @return Response
+     */
+    public function updateContactAndCreateIfNotExists(array $data)
+    {
+        if (isset($data['contacts']) && is_array($data['contacts'])){
+            foreach($data['contacts'] as &$contact){
+                $contact = $this->mapFieldsToIds($contact);
+            }
+        }
```

Comment:
Done. And found one more duplication. Now 3 copies of code packed in utility method. Unfortunately, I can't test affected methods before monday :(